api/app/models/env_canada.py

```diff
@@ -70,14 +65,58 @@ def parse_env_canada_filename(filename):
     prediction_hour = last_part[0][1:]
     prediction_timestamp = model_run_timestamp + \
         datetime.timedelta(hours=int(prediction_hour))
+    return model, variable_name, projection, model_run_timestamp, prediction_timestamp
+
+
+def parse_high_res_model_filename(filename):
+    """ Parse filename for HRDPS grib file to extract metadata """
+    base = os.path.basename(filename)
+    parts = base.split('_')
+    model = '_'.join([parts[1], parts[2]])
+    variable = parts[3]
+    level_type = parts[4]
+    level = parts[5]
+    variable_name = '_'.join(
+        [variable, level_type, level])
+    projection = parts[6]
+    prediction_start = parts[7][:-2]
+    run_time = parts[7][-2:]
+    model_run_timestamp = datetime.datetime(
+        year=int(prediction_start[:4]),
+        month=int(prediction_start[4:6]),
+        day=int(prediction_start[6:8]),
+        hour=int(run_time), tzinfo=datetime.timezone.utc)
+    last_part = parts[8].split('.')
+    prediction_hour = last_part[0][1:4]
+    prediction_timestamp = model_run_timestamp + \
+        datetime.timedelta(hours=int(prediction_hour))
+    return model, variable_name, projection, model_run_timestamp, prediction_timestamp
+
+
+def parse_env_canada_filename(filename):
+    """ Take a grib filename, as per file name nomenclature defined at
+    https://weather.gc.ca/grib/grib2_glb_25km_e.html, and parse into a meaningful object.
+    """
+    # pylint: disable=too-many-locals
+    base = os.path.basename(filename)
+    parts = base.split('_')
+    model = parts[1]
+    if model == 'glb':
+        model, variable_name, projection, model_run_timestamp, prediction_timestamp = \
+            parse_global_model_filename(filename)
+    elif model == 'hrdps':
+        model, variable_name, projection, model_run_timestamp, prediction_timestamp = \
+            parse_high_res_model_filename(filename)
 
     if model == 'glb':
         model_abbreviation = ModelEnum.GDPS
     elif model == 'reg':
         model_abbreviation = ModelEnum.RDPS
+    elif model == 'hrdps_continental':
+        model_abbreviation = ModelEnum.HRDPS
     else:
         raise UnhandledPredictionModelType(
-            'Unhandeled prediction model type found', model)
+            'Unhandled prediction model type found', model)
```

Comment:
I think this can be simplified
if model == 'glb'
elif model == 'hrdps'
elif model == 'reg'
else:
   raise

also which one is the correct one? `hrdps` or `hrdps_continental`

Reply:
Ooof how did I not see that.
Pushed a change with simplified if statements, and fixed it so it's just 'hrdps'